app/code/core/Mage/Core/Block/Template.php

```diff
@@ -205,6 +205,28 @@ public function getShowTemplateHints()
         return self::$_showTemplateHints;
     }
 
+    /**
+     * Retrieve block cache status
+     *
+     * @return  string
+     */
+    private function _getCacheHintStatus(): string
+    {
+        if (!is_null($this->getCacheLifetime())) {
+            return 'green';
+        } else {
+            $_currentParentBlock = $this;
+            $i = 0;
+            while ($i++ < 20 && $_currentParentBlock instanceof Mage_Core_Block_Abstract) {
+                if (!is_null($_currentParentBlock->getCacheLifetime())) {
```

Comment:
How did you set the condition to be limited to 20? Is there a possibility that this value will ever be exceeded?

Reply:
Should it make a maximum of 20 clicks, right?

Reply:
Of course, but is this threshold more or less? I was interested in how you set it to the value of 20.

Reply:
I copied the value and function from AOE, it reaches a maximum of 20 nested blocks